### PR TITLE
Adding support for rich display of Markdown cells in HTML and LaTeX output (redux)

### PR DESCRIPTION
**Pull request by NicholasWMRitchie**
With a fairly small change to the HTML, LaTeX and text display code, it is possible to add rich text output support to cells within a DataFrame. Markdown strings are readily constructed using the default Markdown package and the @md_str macro. It already represents a generalized format definition language that most Julia users are familiar with.

It is already possible to place Markdown.MD cells in DataFrames. However, the display of these cells doesn't represent the richness of which Markdown is capable. With markdown, it is possible to output bold, italicized, headers, links and even images into DataFrame cell outputs.

For me personally, the biggest win is the ability to display functional URLs in a table. This is particularly useful when DataFrames are used in report-type documents like generated by Weave or other literate programming packages. I don't think there is another way to produce functional URL links in DataFrame outputs and this is often useful to detail the provenance of the data in a particular row.

However, other people may find being able to place LaTeX formulae or images in cells even more useful.

### CODE REVIEW

**Comment by NicholasWMRitchie** [comment]
@bkamins I hope this works for you.  I've closed the previous PR.  This whole process has been a learning experience for me.  Hopefully, in the future, I'll be able to handle future PRs more gracefully.  But I am grateful for you patience and I hope these small changes will be of use to DataFrame users.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Thanks for taking the plunge!
   or related issues for context, before opening a new issue
 * Including minimal examples is greatly appreciated
 * If it's a bug, or unexpected behaviour, reproducing on the latest development version
-  (`Pkg.checkout("DataFrames")`) is a good gut check and can streamline the process,
+  (`Pkg.add(name="DataFrames", rev="master")`) is a good gut check and can streamline the process,
```
Comment (bkamins) [inline CONTRIBUTING.md:14]: Please remove this change - you mush have added it by accident from an urelated PR: https://github.com/JuliaData/DataFrames.jl/pull/2387

**Comment by bkamins** [comment]
Thank you!

I have left one small comment. Looks good other than that.

File: test/io.jl [hunk 2]
```diff
@@ -180,6 +203,101 @@ end
...
+            md"",
+            Markdown.parse("∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0" *
+                "∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0"),
+            Markdown.parse("∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ\n"*
+                            "  * ∞7∫αγ\n"*
+                            "  * ∞8∫αγ\n"*
+                            "  * ∞9∫αγ∞0∫α\nγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0"),
+            Markdown.parse("∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0∫α\n"*
+                            "  * γ∞1∫α\n"*
+                            "  * γ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0"),
```
Comment (nalimilan) [inline test/io.jl:270]:
```suggestion
            Markdown.parse("∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0" *
                           "∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0"),
            Markdown.parse("∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ\n"*
                           "  * ∞7∫αγ\n"*
                           "  * ∞8∫αγ\n"*
                           "  * ∞9∫αγ∞0∫α\nγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0"),
            Markdown.parse("∫αγ∞1∫αγ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0∫α\n"*
                           "  * γ∞1∫α\n"*
                           "  * γ∞2∫αγ∞3∫αγ∞4∫αγ∞5∫αγ∞6∫αγ∞7∫αγ∞8∫αγ∞9∫αγ∞0"),
```

**Review by nalimilan — APPROVED** [review]
Thanks @NicholasWMRitchie! I'm not sure why the `Pkg.add` change is visible here since it's also on master, but it should disappear when merging.

**Comment by bkamins** [comment]
I will check if all is OK after the merge

**Comment by bkamins** [comment]
Thank you!

**Comment by bkamins** [comment]
All looks good after merge

**Pull request merged**